changelogs/current.yaml

```diff
@@ -2,6 +2,12 @@ date: Pending
 
 behavior_changes:
 # *Changes that are expected to cause an incompatibility if applicable; deployment changes are likely required*
+- area: wasm
+  change: |
+    The route cache will not be cleared by default if the wasm extension modified the request headers.
+    But it could be done explicitly by call the ``clear_route_cache`` foreign function.
+    This behavior change could be disabled temporarily by setting the runtime flag
+    ``envoy_reloadable_features_wasm_no_automatic_route_refresh`` to false.
```

Comment:
It basically should be replaced by `route_cache_action`. And the default behavior is only clear the route cache when the external plugin require it.

```
    // The default behavior is to clear the route cache only when the
    // :ref:`clear_route_cache <envoy_v3_api_field_service.ext_proc.v3.CommonResponse.clear_route_cache>`
    // field is set in an external processor response.
    DEFAULT = 0;
```

Reply:
@wbpcode Can you introduce this to the xDS spec then instead of the runtime flag. Personally, I think it's OK to change the default to be "do nothing", and allow control planes to set it to "ON_HEADER_OPERATION" to preserve the existing semantics. Istio and other consumers can then control the behavior via xDS which is easier for upgrades.

Would that answer your concerns @PiotrSikora ?

Reply:
@kyessenov Fine. But let me do it step by step. I will remove runtime flag and only keep the abi version check in this PR. Considering abi 0.3.0 is not ready now, so, this change won't break anything.

And at next one or two PR I will add the new API and let the api shepherds to review the API.